### PR TITLE
Pass analyticsID to the elementCall iFrame

### PR DESCRIPTION
**Pull request by toger5**
Signed-off-by: Timo K <timok@element.io>

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

This is needed for: https://github.com/vector-im/element-call/pull/767

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by toger5** [comment]
Do those changes require a test?

**Comment by toger5** [comment]
> Do those changes require a test?

We Internally decided, we want test for it. They are now added.

File: src/PosthogAnalytics.ts [hunk 1]
```diff
@@ -134,7 +134,7 @@ export class PosthogAnalytics {
     private readonly enabled: boolean = false;
     private static _instance = null;
     private platformSuperProperties = {};
-    private static ANALYTICS_EVENT_TYPE = "im.vector.analytics";
+    public static ANALYTICS_EVENT_TYPE = "im.vector.analytics";
```
Comment (robintown) [inline src/PosthogAnalytics.ts:137]:
```suggestion
    public readonly static ANALYTICS_EVENT_TYPE = "im.vector.analytics";
```

File: src/models/Call.ts [hunk 2]
```diff
@@ -637,6 +643,7 @@ export class ElementCall extends Call {
             baseUrl: client.baseUrl,
             lang: getCurrentLanguage().replace("_", "-"),
             fontScale: `${SettingsStore.getValue("baseFontSize") / FontWatcher.DEFAULT_SIZE}`,
+            analyticsID,
```
Comment (robintown) [inline src/models/Call.ts:646]: This could do with a comment explaining that we trust EC with the analytics ID, as discussed internally

**Review by robintown — APPROVED** [review]
Good otherwise

File: src/models/Call.ts (hunk 2 above)
Comment (toger5) [reply src/models/Call.ts:646]: As we discussed. Thanks for reminding me.

Comment (toger5) [reply src/models/Call.ts:650]: Comment was added

**Pull request merged**